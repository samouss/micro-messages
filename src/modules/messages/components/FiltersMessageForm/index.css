.FiltersMessageForm {
  position: fixed;
  display: flex;
  background-color: #D7D7D7;
  width: inherit;
  padding: 8px 10px;
  border-bottom: 1px solid #949494;
}

.FiltersMessageForm__Label {
  font-weight: 600;
  margin-right: 10px;
}
