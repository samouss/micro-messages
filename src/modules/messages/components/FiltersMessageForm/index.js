// @flow

import React from 'react';
import { GroupInput, RadioInput } from 'components/Input';
import Label from 'components/Label';
import './index.css';

import type { Visibility, VisibilityFilter } from '../../types';

export type Props = {
  visibility: VisibilityFilter,
  onChange: VisibilityFilter => void,
}

const FiltersMessageForm = ({ visibility, onChange }: Props) => {
  const onRadioChange = (event: Event & { currentTarget: HTMLInputElement }) => {
    // $FlowFixMe
    onChange(event.currentTarget.value);
  };

  return (
    <form styleName="FiltersMessageForm">
      <span styleName="FiltersMessageForm__Label">
        Visibilit√©:
      </span>

      <GroupInput>
        <RadioInput
          id="all"
          name="visibility"
          onChange={onRadioChange}
          value="all"
          checked={visibility === 'all'}
        >
          Tout
        </RadioInput>

        {(['public', 'private']: Array<Visibility>).map(current => (
          <RadioInput
            key={current}
            id={current}
            name="visibility"
            onChange={onRadioChange}
            value={current}
            checked={visibility === current}
          >
            <Label
              type={current}
            />
          </RadioInput>
        ))}
      </GroupInput>
    </form>
  );
};

export default FiltersMessageForm;
