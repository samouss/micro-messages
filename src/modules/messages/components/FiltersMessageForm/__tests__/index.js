// @flow

import React from 'react';
import { shallow } from 'enzyme';
import FiltersMessageForm from '../index';

import type { Props } from '../index';

describe('<FiltersMessageForm />', () => {
  const defaultProps: Props = {
    visibility: 'all',
    onChange: () => {},
  };

  it('expect to render the component correctly', () => {
    const props: Props = {
      ...defaultProps,
    };

    const component = shallow(
      <FiltersMessageForm
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to render the component correctly with public selected', () => {
    const props: Props = {
      ...defaultProps,
      visibility: 'public',
    };

    const component = shallow(
      <FiltersMessageForm
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });

  it('expect to call onChange on radio change', () => {
    const props: Props = {
      ...defaultProps,
      onChange: jest.fn(),
    };

    const component = shallow(
      <FiltersMessageForm
        {...props}
      />,
    );

    component
      .find('Radio')
      .first()
      .simulate('change');

    component
      .find('Radio')
      .at(1)
      .simulate('change');

    expect(props.onChange).toHaveBeenCalledTimes(2);
  });
});
