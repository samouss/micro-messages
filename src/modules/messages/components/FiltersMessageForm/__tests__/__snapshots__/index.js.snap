// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiltersMessageForm /> expect to render the component correctly 1`] = `
<form
  styleName="FiltersMessageForm"
>
  <span
    styleName="FiltersMessageForm__Label"
  >
    Visibilité:
  </span>
  <Group>
    <Radio
      checked={true}
      id="all"
      name="visibility"
      onChange={[Function]}
      value="all"
    >
      Tout
    </Radio>
    <Radio
      checked={false}
      id="public"
      name="visibility"
      onChange={[Function]}
      value="public"
    >
      <Label
        type="public"
      />
    </Radio>
    <Radio
      checked={false}
      id="private"
      name="visibility"
      onChange={[Function]}
      value="private"
    >
      <Label
        type="private"
      />
    </Radio>
  </Group>
</form>
`;

exports[`<FiltersMessageForm /> expect to render the component correctly with public selected 1`] = `
<form
  styleName="FiltersMessageForm"
>
  <span
    styleName="FiltersMessageForm__Label"
  >
    Visibilité:
  </span>
  <Group>
    <Radio
      checked={false}
      id="all"
      name="visibility"
      onChange={[Function]}
      value="all"
    >
      Tout
    </Radio>
    <Radio
      checked={true}
      id="public"
      name="visibility"
      onChange={[Function]}
      value="public"
    >
      <Label
        type="public"
      />
    </Radio>
    <Radio
      checked={false}
      id="private"
      name="visibility"
      onChange={[Function]}
      value="private"
    >
      <Label
        type="private"
      />
    </Radio>
  </Group>
</form>
`;
